### PR TITLE
Makefile Cleanup

### PR DESCRIPTION
**Pull request by marians**
- removed many variables to make the file more legible
- more comments to foster learning
- renamed USERNAME to GIANTSWARM_USERNAME
- renamed Go-specific variables to make their meaning more clear
- changed base image to `golang:1.4`, since we don’t need cross
compilation and it’s only one third in size
- renamed the generated image with `-go` at the end
- added image removal to “clean”

Tested locally on my machine and on GS, worked fine.

RFR @luebken @puja108

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,67 +1,73 @@
-PROJECT=currentweather
-ORGANIZATION=giantswarm
-REGISTRY = registry.giantswarm.io
-USERNAME := $(shell swarm user)
+# Here we define some variables to be used within
```
Comment (luebken) [inline Makefile]: I would do it a little less verbose. E.g. Variables to be used within the Makefile.

File: Makefile [hunk 2]
```diff
@@ -1,67 +1,73 @@
...
-PROJECT_PATH := $(GOPATH)/src/github.com/$(ORGANIZATION)
-GOOS := linux
-GOARCH := amd64
+GIANTSWARM_USERNAME := $(shell swarm user)
+GO_SOURCE := $(shell find . -name '*.go')
+GO_PATH := $(shell pwd)/.gobuild
+GO_PROJECT_PATH := $(GO_PATH)/src/github.com/giantswarm
 
-.PHONY=all clean deps $(PROJECT) swarm-up docker-build docker-push docker-pull
+# Phony targets are all targets which have names that
```
Comment (luebken) [inline Makefile]: A general Makefile concept. Do we need to explain it here?

File: Makefile [hunk 3]
```diff
@@ -1,67 +1,73 @@
...
+	mkdir -p $(GO_PROJECT_PATH)
+	cd $(GO_PROJECT_PATH) && ln -s ../../../.. currentweather
 
 	# Fetch public packages
-	GOPATH=$(GOPATH) go get -d github.com/$(ORGANIZATION)/$(PROJECT)
+	GOPATH=$(GO_PATH) go get -d github.com/giantswarm/currentweather
 
-$(PROJECT): $(SOURCE) 
-	echo Building for $(GOOS)/$(GOARCH)
+# Compiling the Golang binary for Linux from main.go and libraries.
```
Comment (luebken) [inline Makefile]: Again to verbose: 
Using the official golang container to compile the Golang binary.

File: Makefile [hunk 4]
```diff
@@ -1,67 +1,73 @@
...
+# Starting redis container to run in the background
 docker-run-redis:
-	docker run --name=redis -d redis
-
-docker-run: docker-build
-	docker run --link redis:redis -p 8080:8080 -ti --rm $(REGISTRY)/$(USERNAME)/$(PROJECT)
-
+	docker run --name=currentweather-redis-container -d redis
+
+# Testing your custom-built docker image locally
```
Comment (luebken) [inline Makefile]: Running the container locally. Testing implies automated unit testing.

File: Makefile [hunk 5]
```diff
@@ -1,67 +1,73 @@
...
+docker-run:
+	docker run --link currentweather-redis-container:redis -p 8080:8080 \
+		-ti --rm --name currentweather-go-container \
+		registry.giantswarm.io/$(GIANTSWARM_USERNAME)/currentweather-go
+
+# Pushing the freshly built image to the registry
```
Comment (luebken) [inline Makefile]: Pushing the image to the registry

File: Makefile (hunk 1 above)
Comment (marians) [inline Makefile]: @luebken OK, done

File: Makefile [hunk 6]
```diff
@@ -1,67 +1,73 @@
...
+docker-push:
+	docker push registry.giantswarm.io/$(GIANTSWARM_USERNAME)/currentweather-go
+
+# Starting your application on Giant Swarm.
+# Requires prior pushing to the registry ('make docker-push')
+swarm-up:
+	swarm up swarm.json --var=username=$(GIANTSWARM_USERNAME)
+
+# Removing your application again from Giant Swarm
```
Comment (luebken) [inline Makefile]: Removing your application from Giant Swarm

**Comment by luebken** [comment]
I like the idea of comments in the Makefile but would do it a little less verbose. See inline comments.

File: Makefile (hunk 6 above)
Comment (marians) [inline Makefile]: I actually have a reason to put some more info here. People could actually benefit from this.

**Comment by marians** [comment]
I removed the comment on .PHONY

**Pull request merged**